apps/mochi-web/components/settings/general/MessageModal.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalPortal,
+  ModalTrigger,
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+  TextFieldInput,
+  TextFieldRoot,
+  Typography,
+} from '@mochi-ui/core'
+import React, { useState } from 'react'
+import { Controller, useForm } from 'react-hook-form'
+import { zodResolver } from '@hookform/resolvers/zod'
+import { z } from 'zod'
+import { Message } from './types'
+
+export const actionList = [
+  { key: 'tip', label: 'Tip' },
+  { key: 'payme', label: 'Pay me' },
+  { key: 'paylink', label: 'Pay link' },
+  { key: 'airdrop', label: 'Airdrop' },
+  { key: 'deposit', label: 'Deposit' },
+  { key: 'withdraw', label: 'Withdraw' },
+]
+
+const schema = z.object({
+  action: z.string().min(1, 'This field is required'),
+  message: z.string().min(1, 'This field is required'),
```

Comment:
This looks like a trick to me. Don't we have the `.require()` method?

Reply:
we don't have the .require() method